bokeh/core/has_props.py

```diff
@@ -259,8 +259,9 @@ def __setattr__(self, name, value):
 
         props = sorted(self.properties())
         deprecated = getattr(self, '__deprecated_attributes__', [])
+        descriptor = getattr(self.__class__, name, None)
 
-        if name in props or name in deprecated:
+        if name in props or name in deprecated or (descriptor is not None and descriptor.fset is not None):
             super(HasProps, self).__setattr__(name, value)
```

Comment:
Can you provide context for this change

Reply:
This isn't currently strictly necessary, because I removed commits using this (at least for now). This change is necessary when you have a native property with a setter which isn't deprecated. This `name in deprecated` clause is more of a hack and could be removed in the light of this change. In particular I had a change like this:
```
+    @property
+    def toolbar_location(self):
+        return self.toolbar.location
+
+    @toolbar_location.setter
+    def toolbar_location(self, location):
+        self.toolbar.location = location
```
and setting `plot.toolbar_location = whatever` would result in `AttributeError`.

Reply:
Obviously I need to add a test for this.